Limits co-user

tanya-an opened the pull request:
https://trello.com/c/MWcWt7ri/65-give-to-owner-the-ability-to-manage-co-users-spending-limit-on-account

mesmerze committed:
Merge branch 'jack_molosh_branch' of github.com:iBankRuby/API into cousers

Conflicts:
	app/models/invite.rb
	app/models/role.rb

mesmerze committed:
Merge branch 'backup2507' of github.com:iBankRuby/API into cousers

Conflicts:
	app/models/invite.rb
	app/models/role.rb
	db/schema.rb
	db/seeds.rb

tanya-an committed:
delete avatar (slideNav)

mesmerze committed:
Working on limits/reminders

tanya-an committed:
fix pull

tanya-an committed:
merge branch backup2507

mesmerze committed:
Ref managment contr

tanya-an committed:
added tabs in show accounts

Moleved committed:
Update travis configs

ConstVol committed:
Merge branch 'jack_molosh_branch' of https://github.com/iBankRuby/API into backup2507

ConstVol committed:
Merge branch 'jack_molosh_branch' of https://github.com/iBankRuby/API into backup2507

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into tanya_2

ConstVol committed:
created service to manage account_managemet view

mesmerze committed:
Merge branch 'backup2507' of github.com:iBankRuby/API into limits

Conflicts:
	app/controllers/accounts_controller.rb
	app/controllers/invites_controller.rb
	app/controllers/management_controller.rb
	app/models/account.rb
	app/models/role.rb
	app/models/rule.rb
	db/schema.rb
	db/seeds.rb
	spec/controllers/rules_controller_spec.rb

tanya-an committed:
changed view templates, buttons, styles

tanya-an committed:
Merge branch 'jack_molosh_branch' of https://github.com/iBankRuby/API into tanya_2

mesmerze committed:
Repair db renaming

ConstVol committed:
co-user can't invite anybody to acc, some fixes

tanya-an committed:
edited rules/edit

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into tanya_2

mesmerze committed:
Writing tests

ConstVol committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into backup2507

tanya-an committed:
Added profile owner's name

added: fields to Devise User Model, owner's name in nav

tanya-an committed:
fixed rubocop style warnings

mesmerze committed:
Limits can be viewed, limits tested

tanya-an committed:
Added links to pages

Fix buttons "Back" to be more informational (to show where u're gonna be redirected)

tanya-an committed:
fixed invites and rules controllers

ConstVol committed:
account managemet changes: account user can see his/her role type when managing account

ConstVol committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into backup2507

tanya-an committed:
added mailer

ConstVol committed:
new transaction logic for co-users, added co-users page for owner

mesmerze committed:
Add Valid THRU to account

Add valid IBAN, add Valid THRU to accounts, add check before
each transaction for day and month expiration of account.

ConstVol committed:
Merge branch 'tanya_2' of https://github.com/iBankRuby/API into backup2507

tanya-an committed:
added mailer

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into tanya_2

tanya-an committed:
fix user.rb

tanya-an committed:
Merge branch 'master' of https://github.com/iBankRuby/API into tanya_2

ConstVol committed:
Merge branch 'master' of https://github.com/iBankRuby/API into backup2507

mesmerze committed:
Merge branch 'master' into limits

repair tests

Conflicts:
	app/controllers/accounts_controller.rb
	app/controllers/invites_controller.rb
	app/models/invite.rb
	app/views/accounts/show.html.haml
	db/schema.rb

ConstVol committed:
added transaction over limit confirmation

tanya-an committed:
edited limits

ConstVol committed:
Merge branch 'tanya_2' of https://github.com/iBankRuby/API into backup2507

Moleved committed:
Add exceeding request model, table and associations

Moleved committed:
Add exceeding request controller and test it

mesmerze committed:
Merge branch 'backup2507' of github.com:iBankRuby/API into limits

Repair tests

Conflicts:
	Gemfile
	app/controllers/accounts_controller.rb
	app/controllers/invites_controller.rb
	app/views/accounts/show.html.haml
	config/routes.rb

Moleved committed:
Changing the code according to the style guide requirements

ConstVol committed:
merge fixes

tanya-an committed:
edited "new account"

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into tanya_2

mesmerze committed:
Repair invites for creation limits

ConstVol committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into backup2507

Moleved committed:
Simplify custom validation method

Moleved committed:
Repair mailer and remove unnecessary sendgrid-actionmailer gem

mesmerze committed:
Writing test, remove redundant files, repair logic

ConstVol committed:
specs for management_controller, co_users_controller

ConstVol committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into backup2507

Moleved committed:
Refactor custom validation

mesmerze committed:
Repair views and controllers after wrong deletion

Moleved committed:
Refactor account manager service

ConstVol committed:
co_users_controller, management_controller, transaction_creator refactoring for reek

ConstVol committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into backup2507

mesmerze committed:
Implement soft delete function on accounts

Moleved committed:
Resolve gems version CI error

Moleved committed:
Solve CI bundler version error

ConstVol committed:
management bugfixing

ConstVol committed:
more management bugfixes

tanya-an committed:
added "edit limits" form on co_users page

ConstVol committed:
changes for invitation update: default rule and limit

tanya-an committed:
added a limit setting for co-user

tanya-an committed:
fixed form value

tanya-an committed:
edited: reminder, view "account"

ConstVol committed:
specs for co_users_controller

tanya-an committed:
added redirect

tanya-an committed:
Merge branch 'spendinglimits' of https://github.com/iBankRuby/API into spendinglimits

tanya-an committed:
added registrations_controllers_spec

ConstVol committed:
registrations_controller_spec fixes

mesmerze committed:
Implement soft delete.Partial imeplement of freeze

Conflicts:
	Gemfile.lock
	app/models/account.rb
	app/models/account_user.rb
	app/views/accounts/show.html.haml
	config/routes.rb
	db/schema.rb

tanya-an committed:
add rules form to invite view

tanya-an committed:
Merge branch 'spendinglimits' of https://github.com/iBankRuby/API into spendinglimits

tanya-an committed:
fix invite view

Commit:
add cancancan

tanya-an committed:
Hide ids in routes

ConstVol committed:
new transactions view on account page, some refactoring

ConstVol committed:
Merge branch 'spendinglimits' of https://github.com/iBankRuby/API into backup2507

Moleved committed:
Add exceeding request confirmation function and join it with request sending function

Moleved committed:
Fix specs

mesmerze committed:
Merge branch 'spendinglimits'

Conflicts:
	app/controllers/registrations_controller.rb
	app/models/account.rb
	app/services/account_manager.rb
	db/schema.rb

mesmerze committed:
Merge branch 'backup2507' of github.com:iBankRuby/API into limits

Commit:
authorizations for accounts

mesmerze committed:
Repair soft deleted transactions view

Commit:
move forgery in gemfile, delete junk migration

swindler commented on app/controllers/rules_controller.rb:
```diff
@@ -1,48 +1,48 @@
+#   def new
+#     @rule = Rule.new
+#   end
+#
```
Remove files if they are not needed, don't leave commented. You can always restore using history

swindler commented on app/services/transaction_creator.rb:
```diff
@@ -6,36 +6,70 @@ class TransactionCreator
+    expired = account_to.valid_thru
+    return false if @params[:day] != expired.strftime('%d')
+    return false if @params[:month] != expired.strftime('%m')
+    true
```
rewrite in one line `@params[:day] == expired.strftime('%d') || @params[:month] == expired.strftime('%m')` or similar, not sure with logics here. Probably better use timestamps and compare them

ConstVol committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into backup2507

ConstVol committed:
model account_user fix to delete co-user properly

mesmerze committed:
Repair soft deleted relations

mesmerze committed:
Merge branch 'lockthedoor' into limits

Conflicts:
	Gemfile
	Gemfile.lock
	app/models/rule.rb
	app/views/accounts/index.html.haml
	app/views/accounts/show.html.haml

tanya-an committed:
added input checkbox "movable"

Moleved committed:
Add sending invite message

Moleved committed:
Fix bugs

ConstVol committed:
account_transactions index - all transactions of account for owner, sidekiq worker (not sure if we need it), co-user can't see balance of account only his reminder

tanya-an committed:
Merge branch 'jack_molosh_branch' of https://github.com/iBankRuby/API into spendinglimits

mesmerze committed:
Repair for heroku

mesmerze committed:
Repair for heroku

tanya-an committed:
fixed invites

mesmerze committed:
Repair for Heroku

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

mesmerze committed:
Mailer for Heroku

Commit:
html5 validations for invite form

tanya-an committed:
fixed routes, mailer

mesmerze committed:
Merge branch 'spendinglimits' into limits

Conflicts:
	app/views/accounts/show.html.haml

mesmerze committed:
Repair for heroku

mesmerze committed:
Repair for Heroku

mesmerze committed:
Another repair for Heroku

mesmerze committed:
View repair for heroku

mesmerze committed:
Repair for Heroku

mesmerze committed:
Repair

mesmerze committed:
touch js files

tanya-an committed:
edited view

tanya-an committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into spendinglimits

Commit:
Merge remote-tracking branch 'origin/pushittothelimit' into limits

ConstVol committed:
limit's reminder update fix

tanya-an committed:
fix view

ConstVol committed:
Merge branch 'spendinglimits' of https://github.com/iBankRuby/API into backup2507

Commit:
add app helpers

Commit:
fix seeds, add glossary, disable local mailer, remove junk helpers etc.

Commit:
fix buttons

tanya-an committed:
fix js

tanya-an committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into spendinglimits

ConstVol committed:
statistics builder service

mesmerze committed:
Change transactions model, add migr for state machine

tanya-an committed:
deleted materialize

mesmerze committed:
Merge branch 'limits' of github.com:iBankRuby/API into limits

Conflicts:
	Gemfile
	README.md

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

ConstVol committed:
clean_params method changes

tanya-an committed:
added view statistics

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

mesmerze committed:
Add statuses to transactions

tanya-an committed:
fixed navbar

ConstVol committed:
clean_params must be hash, method for created_at: range

ConstVol committed:
Merge branch 'spendinglimits' of https://github.com/iBankRuby/API into backup2507

ConstVol committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into backup2507

mesmerze committed:
Some changes in routing and reminder replenishment

mesmerze committed:
Add balances to transactions. Add updates for them

ConstVol committed:
some style changes, incoming transactions for statistics

ConstVol committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into backup2507

ConstVol committed:
show user's roles for each account in list on accounts#index, show all account's users and there roles in statsistics

mesmerze committed:
Add cancel from couser, add delayed job gem for stats

mesmerze committed:
Merge branch 'backup2507' of github.com:iBankRuby/API into limits

ConstVol committed:
logout

mesmerze committed:
Merge branch 'backup2507' of github.com:iBankRuby/API into limits

tanya-an committed:
fixed errors

ConstVol committed:
account soft delete error fix

tanya-an committed:
fixed view

Moleved committed:
Repair exceeding request and refactor code base

Moleved committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into limits_copy

mesmerze committed:
Add delayed job for transactions, add soft del to rules

mesmerze committed:
Merge branch 'limits_copy' of github.com:iBankRuby/API into limits

Conflicts:
	app/models/user.rb

Moleved committed:
Merge branch 'spendinglimits' of https://github.com/iBankRuby/API into limits_copy

mesmerze committed:
Merge branch 'spendinglimits' of github.com:iBankRuby/API into limits

Conflicts:
	app/controllers/invites_controller.rb
	app/views/accounts/index.html.haml
	app/views/accounts/show.html.haml
	app/views/co_users/show.html.haml

Moleved committed:
Fix tab simbols

Moleved committed:
Fix rule params within invites controller

tanya-an committed:
fixed view

mesmerze committed:
Delayed job for trans statuses

mesmerze committed:
Delayed job for trans statuses

mesmerze committed:
Delayed job for trans statuses

tanya-an committed:
fixed view

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

tanya-an committed:
fixed view statistics

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

mesmerze committed:
Merge branch 'backup2507' into limits

ConstVol committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into backup2507

Moleved committed:
Fix bugs

mesmerze committed:
Improve trans-s statuses

Moleved committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into limits_copy

mesmerze committed:
Repair couser deletion

Moleved committed:
Repair exceeding request

Moleved committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into limits_copy

mesmerze committed:
Repair transactions expire job

Moleved committed:
Change dependencies for invites table

mesmerze committed:
Merge branch 'limits_copy' of github.com:iBankRuby/API into limits

mesmerze committed:
Change redirects for mailer

mesmerze committed:
Repair js

tanya-an committed:
fixed statistics view

tanya-an committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into spendinglimits

mesmerze committed:
Invectigating where js has died

mesmerze committed:
Repair js, remove invite/rules migrations

mesmerze committed:
Expired transactions on sidekiq

AlexeyAndrianov committed:
fixer notices to be alert for negative actions and events

ConstVol committed:
trans_tracker to reflect transaction's status changes

tanya-an committed:
fix js

tanya-an committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into spendinglimits

Moleved committed:
Repair cancancan gem

Moleved committed:
Add custom methods to cancancan

Moleved committed:
Fix routes

mesmerze committed:
Implement invite statuses, expire them throug sidekiq

AlexeyAndrianov committed:
added edit page for registration info

Moleved committed:
Remove magical comments

mesmerze committed:
Fix invites deletion, fix rules deletion on reject

fix rules deletion on expire, fix invites list

tanya-an committed:
fixed view statistics

mesmerze committed:
Fix bug

If owner delete couser and couser watch owners account
site throw 500. Fixed by catch exception, redirect to
account list

tanya-an committed:
fix balans on accounts view(co-user)

tanya-an committed:
Merge branch 'andrianov2108' of https://github.com/iBankRuby/API into spendinglimits

Moleved committed:
Simplify owner accounts excange

Moleved committed:
Catch record not found error

mesmerze committed:
Merge branch 'limits_copy' into limits

Conflicts:
	app/controllers/accounts_controller.rb
	app/views/accounts/show.html.haml
	db/schema.rb
	db/seeds.rb

tanya-an committed:
removed the balance of the owner from co-user

ConstVol committed:
service for tracking transaction's status changes

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

ConstVol committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into backup2507

mesmerze committed:
Partially implement of simplified transactions

ConstVol committed:
cause of transaction's status changes

Moleved committed:
Replace replenish function to account show

mesmerze committed:
Merge branch 'limits_copy' into limits

Conflicts:
	app/models/user.rb
	app/views/accounts/_modal.html.haml
	app/views/accounts/index.html.haml

mesmerze committed:
Implement simplified transactions

tanya-an committed:
added details transactions

tanya-an committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into spendinglimits

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

Moleved committed:
Repair cancancan

mesmerze committed:
add cancel option to invites

ConstVol committed:
invites tracker to track invite's status changes

Moleved committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into limits_copy

ConstVol committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into backup2507

Moleved committed:
Create additional invite validation

Moleved committed:
Fix invite validation bug with class method

mesmerze committed:
Statuses in exceeding_requests

mesmerze committed:
Merge branch 'limits_copy' of github.com:iBankRuby/API into limits

Conflicts:
	db/schema.rb

tanya-an committed:
fixed details-form on statistics view

ConstVol committed:
tracking invite's status changes with invites_tracking service

ConstVol committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into backup2507

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

ConstVol committed:
something

ConstVol committed:
fixed redirect for co-user after quiting account, new cause for leaving

tanya-an committed:
fixed style

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

ConstVol committed:
lot of changes

mesmerze committed:
Repair invites creation validation

mesmerze committed:
Merge branch 'backup2507' into limits

Conflicts:
	app/models/invite.rb

tanya-an committed:
added invites list in statistics and fixed bottom

tanya-an committed:
fixed tabs in page "accounts"

mesmerze committed:
Fix expiring invites, inc trans for couser

tanya-an committed:
edited view statistics

ConstVol committed:
dump.rdb to .gitignore

tanya-an committed:
Merge branch 'limits' of https://github.com/iBankRuby/API into spendinglimits

ConstVol committed:
Merge branch 'spendinglimits' of https://github.com/iBankRuby/API into backup2507

tanya-an committed:
fixed registrations_controller_spec

ConstVol committed:
coverage

tanya-an committed:
invites_controller_spec: added action "confirm"

ConstVol committed:
more invites controller coverage

tanya-an committed:
invites_controller_spec: added action "delete"

tanya-an committed:
Merge branch 'backup2507' of https://github.com/iBankRuby/API into spendinglimits

tanya-an committed:
fixed date in statistics
